### PR TITLE
[C20-5108] add new rules

### LINKED ISSUE #7: Propose eslint rule for JSDoc
JSDoc: https://eslint.org/docs/rules/valid-jsdoc

This one is something that we already agreed before, but never actually documented. Is to in the JSDoc `param` arguments always use lower case for primitive types (e.g., `string`, `number`, etc), and upper case for non-primitive types (e.g., `Object`, `Array`).
Other configurations are also possible.

```
/**
 * @param {Array<Object>} messages
 * @param {string} eventType
 * @returns {Object}
 */
```

### LINKED ISSUE #12: new suggested rules
https://eslint.org/docs/rules/comma-dangle
We already have it. I suggest we change it to this,
```js
'comma-dangle': ["warn", {
      "arrays": "always-multiline",
      "objects": "always-multiline",
      "imports": "always-multiline",
      "exports": "always-multiline",
      "functions": "never"
    }],
```

https://eslint.org/docs/rules/no-multi-spaces
warn.

https://eslint.org/docs/rules/require-jsdoc
warn. Always require JSDoc.

https://eslint.org/docs/rules/arrow-parens
warn with "as-needed"

https://eslint.org/docs/rules/curly
Not sure what's best here.

https://eslint.org/docs/rules/brace-style
 ["warn", "1tbs", { "allowSingleLine": true }],

https://eslint.org/docs/rules/space-before-function-paren
Always, warn.

https://eslint.org/docs/rules/no-invalid-this
off this one.

https://eslint.org/docs/rules/no-prototype-builtins
warn

https://eslint.org/docs/rules/no-trailing-spaces
warn

Later on, these will be charged to Errors.
cc @relayr/nodejs

### PR DESCRIPTION
**Pull request by WaleedAshraf**
closes #12, closes #7 

Will merge if there's no objection in the next couple of days.
@relayr/nodejs

### CODE REVIEW

**Comment by WaleedAshraf** [comment]
@relayr/nodejs If you start working on some service,
I'll suggest that you open 2 PRs, both should from master to master. Independent of each other.

- One for your actual task. (Don't bother linting in this one)

- One for fixing the linting. Run `./node_modules/.bin/eslint ./ --fix` to fix all issue. Don't change anything else in this PR.

File: index.js [hunk 1]
```diff
@@ -3,23 +3,54 @@ module.exports = {
...
     'no-multiple-empty-lines': 'off',
+    'curly': 'off',
 
     // warn
     'camelcase': 'warn',
-    'comma-dangle': ['warn', 'always-multiline'],
+    'one-var': ['warn', 'consecutive'],
+    'valid-jsdoc': ['warn', {
+      'requireParamType': true,
+      'prefer': {
```
Comment (ssncferreira) [inline index.js:19]: Maybe also add for `return`, because in some places we use `return` while in others `returns`.

File: index.js [hunk 2]
```diff
@@ -3,23 +3,54 @@ module.exports = {
...
+        'arg': 'param',
+        'argument': 'param',
+      },
+      'preferType': { 
+        'Boolean': 'boolean',
+        'Number': 'number',
+        'String': 'string',
+        'object': 'Object',
+        'array': 'Array',
```
Comment (ssncferreira) [inline index.js:28]: Did you checked @19shubham11 comment about Array of type: https://github.com/relayr/eslint-config-relayr/issues/7#issuecomment-645920766
Do you know if there is also a rule for it?

File: index.js (hunk 1 above)
Comment (WaleedAshraf) [reply index.js:19]: I think both are fine? Should we enforce one of them?

File: index.js (hunk 2 above)
Comment (WaleedAshraf) [reply index.js:28]: I saw what Shubham suggested. I added a reply there. Both formats are valid.

File: index.js (hunk 1 above)
Comment (ssncferreira) [reply index.js:19]: I'm in favour of having consistent code 🤓 so I would prefer to force one of them.
But if that is too much, I'm ok with allowing both 👍

Comment (WaleedAshraf) [reply index.js:19]: let discuss that in a separate issue. I'll merge this for now. We can always add more strict rules.

File: index.js (hunk 2 above)
Comment (WaleedAshraf) [reply index.js:28]: let discuss that in a separate issue. I'll merge this for now. We can always add more strict rules.

**Pull request merged**